controllers/vgmanager_daemonset.go

```diff
@@ -0,0 +1,186 @@
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	hostContainerPropagation  = corev1.MountPropagationHostToContainer
+	directoryHostPath         = corev1.HostPathDirectory
+	HostPathDirectoryOrCreate = corev1.HostPathDirectoryOrCreate
+
+	LVMdVolName   = "lvmd-conf"
+	UdevVolName   = "run-udev"
+	DevDirVolName = "device-dir"
+	SysVolName    = "sys"
+
+	LVMdDir      = "/mnt/lvmd/"
+	LVMdConfPath = "/mnt/lvmd/lvmd.conf"
+	devDirPath   = "/dev"
+	udevPath     = "/run/udev"
+	sysPath      = "/sys"
+
+	// LVMDConfVol is the corev1.Volume definition for the directory on host ("/mnt/lvmd/") for storing
+	// the lvmd.conf file
+	LVMDConfVol = corev1.Volume{
+		Name: LVMdVolName,
+		VolumeSource: corev1.VolumeSource{
+			HostPath: &corev1.HostPathVolumeSource{
+				Path: LVMdDir,
+				Type: &HostPathDirectoryOrCreate,
+			},
+		},
+	}
+
+	// LVMDConfVolMount is the corresponding mount for LVMDConfVol
+	LVMDConfVolMount = corev1.VolumeMount{
+		Name:             LVMdVolName,
+		MountPath:        LVMdDir,
+		MountPropagation: &hostContainerPropagation,
+	}
+
+	// DevHostDirVol  is the corev1.Volume definition for the "/dev" bind mount used to
+	// list block devices.
+	// DevMount is the corresponding mount
+	DevHostDirVol = corev1.Volume{
+		Name: DevDirVolName,
+		VolumeSource: corev1.VolumeSource{
+			HostPath: &corev1.HostPathVolumeSource{
+				Path: devDirPath,
+				Type: &directoryHostPath,
+			},
+		},
+	}
+
+	// DevMount is the corresponding mount for DevHostDirVol
+	DevHostDirVolMount = corev1.VolumeMount{
+		Name:             DevDirVolName,
+		MountPath:        devDirPath,
+		MountPropagation: &hostContainerPropagation,
+	}
+
+	// UDevHostDirVol is the corev1.Volume definition for the
+	// "/run/udev" host bind-mount. This helps lsblk give more accurate output.
+	// UDevMount is the corresponding mount
+	UDevHostDirVol = corev1.Volume{
+		Name: UdevVolName,
+		VolumeSource: corev1.VolumeSource{
+			HostPath: &corev1.HostPathVolumeSource{Path: udevPath},
+		},
+	}
+
+	// UDevMount is the corresponding mount for UDevHostDirVol
+	UDevHostDirVolMount = corev1.VolumeMount{
+		Name:             UdevVolName,
+		MountPath:        udevPath,
+		MountPropagation: &hostContainerPropagation,
+	}
+
+	// SysHostDirVol is the corev1.Volume definition for the
+	// "/sys" host bind-mount. This helps discover information about blockd devices
+	SysHostDirVol = corev1.Volume{
+		Name: SysVolName,
+		VolumeSource: corev1.VolumeSource{
+			HostPath: &corev1.HostPathVolumeSource{Path: sysPath},
+		},
+	}
+
+	// SysMount is the corresponding mount for SysHostDirVol
+	SysHostDirVolMount = corev1.VolumeMount{
+		Name:             SysVolName,
+		MountPath:        sysPath,
+		MountPropagation: &hostContainerPropagation,
+	}
+)
+
+// newVGManagerDaemonset returns the desired vgmanager daemonset for a given LVMCluster
+func newVGManagerDaemonset(lvmCluster lvmv1alpha1.LVMCluster) appsv1.DaemonSet {
```

Comment:
In my view, probably we should add all the resources needed to create the daemonSet in this function, and not trust completelly  in the createOrUpdate function to complete the daemonset definition.

Reply:
Agreed. @sp98 , why are the nodeselector and tolerations not part of this function?

Reply:
That was a bug. Now updating the nodeSelector, toleration and ownerreferences after this function is called.

Reply:
It is better to set the tolerations, nodeSelector and serviceAccount in this function so the returned daemonSet is complete.

Reply:
done.